From 623b5f2703b6379acd0ab85866e9b798dfd12f15 Mon Sep 17 00:00:00 2001
From: Francisco Molina <femolina@uc.cl>
Date: Tue, 8 Mar 2022 16:39:35 +0100
Subject: [PATCH 6/6] lib/rng_math: remove assert, PR is <0

---
 lib/rng_math/src/rng_math.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/rng_math/src/rng_math.c b/lib/rng_math/src/rng_math.c
index 4a6ca0f..cb597fd 100644
--- a/lib/rng_math/src/rng_math.c
+++ b/lib/rng_math/src/rng_math.c
@@ -146,7 +146,6 @@ calc_tof_sym_ds(uint32_t first_response_timestamp, uint32_t first_request_timest
 float
 uwb_rng_path_loss(float Pt, float G, float fc, float R){
     float Pr = Pt + 2 * G + 20 * log10(299792458.0l/1.000293l) - 20 * log10(4 * M_PI * fc * R);
-    assert(Pr >= 0);
     return Pr;
 }
 #endif
-- 
2.32.0
