From 27be50625bed86f7b2d3cbedc373e0c26f276e93 Mon Sep 17 00:00:00 2001
From: Francisco Molina <femolina@uc.cl>
Date: Mon, 9 Aug 2021 14:14:48 +0200
Subject: [PATCH] src/crypto/tinycrypt: add crypto_gen_keypair

---
 src/crypto/tinycrypt.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/src/crypto/tinycrypt.c b/src/crypto/tinycrypt.c
index ee1f6e9..0267851 100644
--- a/src/crypto/tinycrypt.c
+++ b/src/crypto/tinycrypt.c
@@ -12,10 +12,20 @@
 #include "tinycrypt/ccm_mode.h"
 #include "c25519/c25519.h"
 #include "c25519/edsign.h"
+#include "random.h"
 
 int crypt_gen_keypair(cose_curve_t crv, cose_key_t *key) {
-    (void) crv;
-    (void) key;
+    if (crv == COSE_EC_CURVE_X25519) {
+        cose_key_init(key);
+        random_bytes(key->d, C25519_EXPONENT_SIZE);
+        c25519_prepare(key->d);
+        c25519_smult(key->x, c25519_base_x, key->d);
+        key->dLen = C25519_EXPONENT_SIZE;
+        key->xLen = C25519_EXPONENT_SIZE;
+        key->crv = COSE_EC_CURVE_X25519;
+        key->kty = COSE_KTY_OCTET;
+        return EDHOC_SUCCESS;
+    }
     return EDHOC_ERR_RANDOMNESS;
 }
 
-- 
2.32.0

